File: library/src/test/java/me/proxer/library/enums/RelationshipStatusTest.java
package me.proxer.library.enums;

import me.proxer.library.ProxerTest;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * @author Ruben Gees
 */
class RelationshipStatusTest extends ProxerTest {

    @Test
    void testDefault() throws IOException {
        final RelationshipStatus relationshipStatus = api.moshi().adapter(RelationshipStatus.class)
                .fromJson("\"single\"");

        assertThat(relationshipStatus).isSameAs(RelationshipStatus.SINGLE);
    }

    @Test
    void testFallback() throws IOException {
        final RelationshipStatus relationshipStatus = api.moshi().adapter(RelationshipStatus.class)
                .fromJson("\"xyz\"");

        assertThat(relationshipStatus).isSameAs(RelationshipStatus.UNKNOWN);
    }
}

File: library/src/test/kotlin/me/proxer/library/enums/RelationshipStatusTest.kt
package me.proxer.library.enums

import com.serjltt.moshi.adapters.FallbackEnum
import com.squareup.moshi.Moshi
import org.amshove.kluent.shouldBe
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class RelationshipStatusTest {

    private val adapter = Moshi.Builder()
        .add(FallbackEnum.ADAPTER_FACTORY)
        .build()
        .adapter(RelationshipStatus::class.java)

    @Test
    fun testDefault() {
        adapter.fromJson("\"single\"") shouldBe RelationshipStatus.SINGLE
    }

    @Test
    fun testFallback() {
        adapter.fromJson("\"xyz\"") shouldBe RelationshipStatus.UNKNOWN
    }
}
